File: ResourceManager/Helpers/ViewPager.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace ResourceManager.Helpers
    Public Class ViewPager(Of T As Class)
        Private _SelectedPageItem As Integer
        Private _totalRecords As Integer
        Private _totalPages As Integer
        Private _pageSize As Integer
        Public Sub New(ByVal contents As IEnumerable(Of T), ByVal Optional selectedPageItem As Integer = 1, ByVal Optional selectedRowsPerPage As String = "10")
            Me.Contents = contents
            ViewContent = [Goto](selectedPageItem, selectedRowsPerPage)
        End Sub
        Public Function [Goto](ByVal Optional selectedPageItem As Integer = 1, ByVal Optional selectedRowsPerPage As String = "10") As IEnumerable(Of T)
            _totalRecords = Contents.Count()
            Me.SelectedPageItem = selectedPageItem
            _pageSize = If(selectedRowsPerPage.Equals("all"), If(_totalRecords < 1, 1, _totalRecords), Convert.ToInt32(selectedRowsPerPage))
            _totalPages = If(_totalRecords < 1, 1, _totalRecords / _pageSize)
            If _totalRecords Mod _pageSize > 0 Then _totalPages += 1
            Dim pageItems As IList(Of Integer) = New List(Of Integer)()
            For i = 1 To _totalPages
                pageItems.Add(i)
            Next
            Me.PageItems = pageItems
            If Me.SelectedPageItem > _totalPages Then
                Me.SelectedPageItem = _totalPages
            End If
            Return Contents.Skip((Me.SelectedPageItem - 1) * _pageSize).Take(_pageSize)
        End Function

        Public Property SelectedPageItem As Integer
            Get
                Return _SelectedPageItem
            End Get
            Private Set(ByVal value As Integer)
                _SelectedPageItem = value
            End Set
        End Property
        Public Property PageItems As IEnumerable(Of Integer) = New List(Of Integer)()
        Public Property ViewContent As IEnumerable(Of T)
        Public ReadOnly Property Contents As IEnumerable(Of T)
    End Class
End Namespace

File: ResourceManager/Helpers/ViewPager.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ResourceManager.Helpers
{
    public class ViewPager<T> where T : class
    {
        private int _totalRecords;
        private int _totalPages;
        private int _pageSize;
        public ViewPager(IEnumerable<T> contents, int selectedPageItem = 1, string selectedRowsPerPage = "10")
        {
            Contents=contents;
            ViewContent = Goto(selectedPageItem, selectedRowsPerPage);
        }
        public IEnumerable<T> Goto(int selectedPageItem = 1, string selectedRowsPerPage = "10")
        {
            _totalRecords = Contents.Count();
            SelectedPageItem = selectedPageItem;
            _pageSize = selectedRowsPerPage.Equals("all") ? (_totalRecords < 1 ? 1 : _totalRecords) : Convert.ToInt32(selectedRowsPerPage);
            _totalPages = _totalRecords < 1 ? 1 : _totalRecords / _pageSize;
            if (_totalRecords % _pageSize > 0) _totalPages++;
            IList<int> pageItems = new List<int>();
            for (int i = 1; i <= _totalPages; i++)pageItems.Add(i);
            PageItems = pageItems;
            if (SelectedPageItem > _totalPages)
            {
                SelectedPageItem = _totalPages;
            }
            return Contents.Skip((SelectedPageItem - 1) * _pageSize).Take(_pageSize);
        }
        public int SelectedPageItem { get; private set; }
        public IEnumerable<int> PageItems { get; set; } = new List<int>();
        public IEnumerable<T> ViewContent { get; set; }
        public IEnumerable<T> Contents { get; }
    }
}
